packages/calcite-components/src/components/panel/panel.e2e.ts

```diff
@@ -129,6 +133,49 @@ describe("calcite-panel", () => {
     expect(await container.isVisible()).toBe(false);
   });
 
+  it("should handle rejected 'beforeClose' promise'", async () => {
+    const page = await newE2EPage();
+
+    const mockCallBack = jest.fn().mockReturnValue(() => Promise.reject());
+    await page.exposeFunction("beforeClose", mockCallBack);
+
+    await page.setContent(`<calcite-panel closable></calcite-panel>`);
+
+    await page.$eval(
+      "calcite-panel",
+      (el: HTMLCalcitePanelElement) =>
+        (el.beforeClose = (window as typeof window & Pick<typeof el, "beforeClose">).beforeClose),
```

Comment:
For consistency, could you use our [custom utility type](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/tests/utils.ts#L9) to define custom props on `window` ([example](https://github.com/Esri/calcite-design-system/blob/60ab0dcf5e571ede472f62f751ef4a380181584f/packages/calcite-components/src/components/button/button.e2e.ts#L658-L660))?

Reply:
Added as part of https://github.com/Esri/calcite-design-system/pull/9778